web/scripts/rate_all.py

```diff
@@ -22,42 +23,59 @@ def sanitized_games(games):
     print("Cleaning games...")
     for g in games:
         if g.white.user_id is None or g.black.user_id is None:
-            print('  ', g) #should probably strip them.
+            print('No ids        :  ', g) #should probably strip them from the db.
+            pass
         elif g.handicap > 1 and (-1 > g.komi > 1):
-            print('  ', g)
+            #print('bad komi      :  ', g)
+            pass
         elif g.handicap > 6:
             continue
         elif g.komi < 0 or g.komi > 8.6:
-            print('  ', g)
-        elif not (g.result.startswith('W+') or g.result.startswith('B+')):
-            print('  ', g)
+            #print('bad komi      :  ', g)
+            pass
+        elif not (g.result.startswith('W') or g.result.startswith('B')):
+            print('unknown result:  ', g)
+            pass
         else:
-            # Vector of result_tuples.  Just what we need to compute stuff...
-            g_vec.append( (g.white.user_id, 
-                         g.black.user_id,
+            # Vector of result_tuples.  Just what we need to compute ratings...
+            g_vec.append( (g.white.id, 
```

Comment:
yeah it's getting a little unwieldy, isn't it.  This is basically the nut, though; each iteration it gets shuffled and re-stepped through, so i'm assuming it'll help to keep it simpler.  My guess is making it a numpy array is probably gonna get a speedup.